cmake/generic_toolchain.cmake

```diff
@@ -47,7 +47,10 @@ set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT} CACHE STRING "Zephyr to
 assert(ZEPHYR_TOOLCHAIN_VARIANT "Zephyr toolchain variant invalid: please set the ZEPHYR_TOOLCHAIN_VARIANT-variable")
 
 # Pick host system's toolchain if we are targeting posix
-if((${ARCH} STREQUAL "posix") OR (${ARCH} STREQUAL "x86_64"))
+# or x86_64 (but not using xtools)
+if((${ARCH} STREQUAL "posix") OR
+   ((${ARCH} STREQUAL "x86_64")
+    AND (NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xtools")))
```

Comment:
Once the x86_64 compiler is in the SDK, we can actually remove this bit and just use ZEPHYR_TOOLCHAIN_VARIANT=host as it was originally intended.

Maybe revert the sense here and force the toolchain to host only if it started at "zephyr" to begin with, for greater compatibility?  Probably not a big deal.

Reply:
Once the SDK has the x86_64 toolchain, we can remove the the "x86_64 and !xtools" part as x86_64 will be built using the toolchain in SDK.

Reply:
> as x86_64 will be built using the toolchain in SDK.

again this seems to imply that the number of toolchain options will be reduced. Is this the case?

Reply:
The intention is to test the sdk-ng built x86_64 toolchain before we actually include it in the SDK, hence enabling it for xtools, for now. This also allows developers to have a common base for comparison if there is a bug report. Once we have the x86_64 toolchain in the official SDK, the host toolchain will only be used for when ARCH==posix.

Does this answer your question?

Reply:
> Once we have the x86_64 toolchain in the official SDK, the host toolchain will only be used for when ARCH==posix.

AFAIK the choice of the toolchain is made by the end-user with the ZEPHYR_TOOLCHAIN_VARIANT variable. Do you plan to remove that choice?

Reply:
No. The change inside cmake/toolchain/xtools will stay.

Or, do you want to use the host toolchain to build for x86_64 even if `ZEPHYR_TOOLCHAIN_VARIANT=xtools`?

Reply:
@marc-hb: it's sort of the converse.  Right now, there is no SDK support for x86_64 and posix, so we effectively force ZEPHYR_TOOLCHAIN_VARIANT to be "host" for those platforms even if it started out as "zephyr".

Reply:
> we effectively force ZEPHYR_TOOLCHAIN_VARIANT to be "host" for those platforms even if it started out as "zephyr".

Thanks, I didn't expect such a hack. I guess it was added to keep CI simple while adding some coverage?

So because the current (and too short) commit message/description assumes knowledge of this hack it almost sounds like it's adding such a restriction/hack when it will actually allow removing one...